src/clients/BundleDataClient/BundleDataClient.ts

```diff
@@ -1095,37 +1072,132 @@ export class BundleDataClient {
+                validatedBundleV3Fills.push({
+                  ...fill,
+                  quoteTimestamp: deposit.quoteTimestamp,
+                });
+              }
+              return;
+            }
+
+            // If a slow fill request exists in memory, then we know the deposit has not been filled because fills
+            // must follow slow fill requests and we would have seen the fill already if it existed. Therefore,
+            // we can conclude that either the deposit has expired and we need to create a deposit expiry refund, or
+            // we need to create a slow fill leaf for the deposit. The latter should only happen if the slow fill request
+            // took place in a prior bundle otherwise we would have already created a slow fill leaf for it.
+            if (slowFillRequest) {
+              if (_depositIsExpired(deposit)) {
+                updateExpiredDepositsV3(expiredDepositsToRefundV3, deposit);
+              } else if (
+                canRefundPrefills &&
+                slowFillRequest.blockNumber < destinationChainBlockRange[0] &&
+                _canCreateSlowFillLeaf(deposit)
+              ) {
+                validatedBundleSlowFills.push(deposit);
+              }
+              return;
+            }
+
+            // So at this point in the code, there is no fill or slow fill request in memory for this deposit.
+            // We need to check its fill status on-chain to figure out whether to issue a refund or a slow fill leaf.
+            // We can assume at this point that all fills or slow fill requests, if found, were in previous bundles
+            // because the spoke pool client lookback would have returned this entire bundle of events and stored
+            // them into the relay hash dictionary.
+            const fillStatus = await _getFillStatusForDeposit(deposit, destinationChainBlockRange[1]);
+
+            // If deposit was filled, then we need to issue a refund for it.
+            if (fillStatus === FillStatus.Filled) {
+              // We need to find the fill event to issue a refund to the right relayer and repayment chain,
+              // or msg.sender if relayer address is invalid for the repayment chain.
+              const prefill = (await findFillEvent(
+                destinationClient.spokePool,
+                deposit,
+                destinationClient.deploymentBlock,
+                destinationClient.latestBlockSearched
+              )) as unknown as FillWithBlock;
+              const verifiedFill = await verifyFillRepayment(
+                prefill,
+                destinationClient.spokePool.provider,
+                deposit,
+                allChainIds
+              );
+              if (canRefundPrefills && isDefined(verifiedFill) && !isSlowFill(prefill)) {
+                validatedBundleV3Fills.push({
+                  ...prefill,
```

Comment:
I think you need to push verifiedFill here in case the relayer repayment address changed.

Reply:
yep realized this while adding a unit test